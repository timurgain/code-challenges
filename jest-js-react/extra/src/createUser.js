export function createUser({
  name,
  surname,
  age = 18,
  email = '',
}) {
  return {
    type: 'USER',
    name,
    surname,
    password: 'hashed-password',
    age,
    email,
  };
}
