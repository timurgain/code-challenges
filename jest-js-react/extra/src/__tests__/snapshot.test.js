import { createUser } from '../createUser.js';

describe('createUser', () => {
  it('should create a user with name and surname only', () => {
    const user = { name: 'John', surname: 'Doe' };
    expect(createUser(user)).toMatchSnapshot();
  });

  it('should create a user with all the properties', () => {
    const user = {
      name: 'John',
      surname: 'Doe',
      age: 30,
      email: 'e@em.com'
    };
    expect(createUser(user)).toMatchSnapshot();
  });
});
