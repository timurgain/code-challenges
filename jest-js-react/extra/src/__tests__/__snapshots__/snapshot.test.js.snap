// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createUser should create a user with all the properties 1`] = `
{
  "age": 30,
  "email": "e@em.com",
  "name": "John",
  "password": "hashed-password",
  "surname": "Doe",
  "type": "USER",
}
`;

exports[`createUser should create a user with name and surname only 1`] = `
{
  "age": 18,
  "email": "",
  "name": "John",
  "password": "hashed-password",
  "surname": "Doe",
  "type": "USER",
}
`;
